packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -158,157 +124,111 @@ const ImmunizationsForm: React.FC<ImmunizationsFormProps> = ({
+            value={formState.expirationDate}
+            onChange={([date]) => updateSingle('expirationDate', date)}>
+            <DatePickerInput
+              id="date-picker-calendar-id"
+              placeholder="dd/mm/yyyy"
+              labelText={t('expirationDate', 'Expiration Date')}
+              type="text"
+            />
+          </DatePicker>
+        </div>
+        <div className={styles.immunizationSequenceSelect}>
+          <TextInput
+            type="text"
+            id="lotNumber"
+            labelText={t('lotNumber', 'Lot Number')}
+            value={formState.lotNumber}
+            onChange={(evt) => updateSingle('lotNumber', evt.target.value)}
+          />
+        </div>
+        <div className={styles.immunizationSequenceSelect}>
+          <TextInput
+            type="text"
+            id="manufacturer"
+            labelText={t('manufacturer', 'Manufacturer')}
+            value={formState.manufacturer}
+            onChange={(evt) => updateSingle('manufacturer', evt.target.value)}
+          />
+        </div>
+        <div className={styles.immunizationSequenceSelect}>
+          <div
             className={
               enableCreateButtons || enableEditButtons
-                ? 'omrs-btn omrs-filled-action omrs-rounded'
-                : 'omrs-btn omrs-outlined omrs-rounded'
-            }
-            disabled={isViewEditMode ? !enableEditButtons : !enableCreateButtons}>
-            {t('save', 'Save')}
-          </button>
+                ? `${styles.buttonStyles} ${styles.buttonStylesBorder}`
+                : styles.buttonStyles
+            }>
+            <Button
+              type="button"
+              kind="secondary"
+              style={{ width: '50%', marginBottom: '1rem' }}
+              onClick={closeWorkspace}>
+              {t('cancel', 'Cancel')}
+            </Button>
+            <Button
+              type="submit"
+              kind="primary"
+              style={{ width: '50%', marginBottom: '1rem' }}
+              disabled={isViewEditMode ? !enableEditButtons : !enableCreateButtons}>
+              {t('save', 'Save')}
+            </Button>
+          </div>
         </div>
-      </form>
+      </Form>
     );
   }
-
-  const closeForm = (event) => {
-    let userConfirmed: boolean = false;
-    const defaultConfirmMessage = 'There is ongoing work, are you sure you want to close this tab?';
-    const confirmMessage = t('close form confirm message', defaultConfirmMessage);
-    if (formState.formChanged) {
-      userConfirmed = confirm(confirmMessage);
-    }
-
-    if (userConfirmed && formState.formChanged) {
-      entryCancelled();
-      history.push(`/patient/${patientUuid}/chart/immunizations`);
-      closeComponent();
-    } else if (!formState.formChanged) {
-      entryCancelled();
-      history.push(`/patient/${patientUuid}/chart/immunizations`);
-      closeComponent();
-    }
-  };
-
   return <div>{createForm()}</div>;
```

Comment:
Probably better to put the code for `createForm()` component inside here rather than calling the function as so.